aioaws/core.py

```diff
@@ -23,12 +23,108 @@
 _AUTH_ALGORITHM = 'AWS4-HMAC-SHA256'
 
 
+def _aws4_date_stamp(dt: datetime) -> str:
+    return dt.strftime('%Y%m%d')
+
+
+def _aws4_x_amz_date(dt: datetime) -> str:
+    return dt.strftime('%Y%m%dT%H%M%SZ')
+
+
+def _aws4_reduce_signature(key: bytes, msg: str) -> bytes:
+    return hmac.new(key, msg.encode(), hashlib.sha256).digest()
+
+
+class _AWSv4Auth:
+    def __init__(
+        self,
+        aws_secret_key: str,
+        aws_access_key_id: str,
+        region: str,
+        service: str,
+    ) -> None:
+        self.aws_secret_key = aws_secret_key
+        self.aws_access_key_id = aws_access_key_id
+        self.region = region
+        self.service = service
+
+    def auth_headers(
```

Comment:
This logic is very specific and very small changes can break authentication in opaque ways.

Any chance you could move `_AWSv4Auth` below `AwsClient` so the diff is smaller and all this code remains (in git's eyes) unchanged? No worries if not.

Reply:
Agreed. I didn't make any changes FWIW. I moved things around to try to minimize the diff.